src/Microsoft.Health.SqlServer/Features/Schema/Manager/SqlSchemaManager.cs

```diff
@@ -119,7 +119,18 @@ public virtual async Task ApplySchema(MutuallyExclusiveType type, bool force = f
                 throw new SchemaManagerException(string.Format(CultureInfo.CurrentCulture, Resources.SpecifiedVersionNotAvailable, targetVersion));
             }
 
-            await ValidateVersionCompatibility(availableVersions[^1].Id, token).ConfigureAwait(false);
+            // TTL in instance schema table can cause this call to fail on first attempt so allow for retries
+            attemptCount = 1;
```

Comment:
The `onRetry` delegate is providing this information; you don't need to track it yourself